flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskManagerLocationTest.java

```diff
@@ -198,4 +199,18 @@ public void testGetHostname2() {
 			fail(e.getMessage());
 		}
 	}
+
+	@Test
+	public void testNotRetrieveHostName() {
+		InetAddress address = mock(InetAddress.class);
+		when(address.getCanonicalHostName()).thenReturn("worker10");
+		when(address.getHostName()).thenReturn("worker10");
+		when(address.getHostAddress()).thenReturn("127.0.0.1");
```

Comment:
Instead of mocking, could we use `127.0.0.1` which should resolve to `"localhost"` or are these assumption too strong?

Reply:
Hi @tillrohrmann , based on my prior observation, some of the machines might not resolve "127.0.0.1" back to "localhost" due to missing `/etc/hosts` items or other reasons, so I think we'd better explicitly mock the return value to offer more compatibility in various environments.

Reply:
Ok, this sounds like a good argument for keeping the mocking in place.